src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,67 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationFailureException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_DATA_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+    private StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        userNullDataValidation(user);
+        userLoginAndPasswordValidation(user);
+        ageValidation(user);
```

Comment:
The order of those validations is a bit messy, methods should have as isolated and single responsibility as possible, so usually we do group several different fields by a single validation criteria, so let's add couple methods like `validateLogin()` `validatePassword()` `validateAge()` and we can call them in some `validateUser()` instead of what we have here